aiven/client/envdefault.py

```diff
@@ -7,14 +7,21 @@
 """
 
 import os
+from typing import Optional, Union
 
-USER_HOME = os.path.expanduser("~")
+USER_HOME: str = os.path.expanduser("~")
 
-AIVEN_CONFIG_DIR = os.environ.get("AIVEN_CONFIG_DIR", os.path.join(USER_HOME, ".config", "aiven"))
+AIVEN_CONFIG_DIR: Optional[str] = os.environ.get(
+    "AIVEN_CONFIG_DIR", os.path.join(USER_HOME, ".config", "aiven")
+)
 
-AIVEN_AUTH_TOKEN = os.environ.get("AIVEN_AUTH_TOKEN")
-AIVEN_CA_CERT = os.environ.get("AIVEN_CA_CERT")
-AIVEN_CLIENT_CONFIG = os.environ.get("AIVEN_CLIENT_CONFIG", os.path.join(AIVEN_CONFIG_DIR, "aiven-client.json"))
-AIVEN_CREDENTIALS_FILE = os.environ.get("AIVEN_CREDENTIALS_FILE", os.path.join(AIVEN_CONFIG_DIR, "aiven-credentials.json"))
-AIVEN_PROJECT = os.environ.get("AIVEN_PROJECT")
-AIVEN_WEB_URL = os.environ.get("AIVEN_WEB_URL")
+AIVEN_AUTH_TOKEN: Optional[str] = os.environ.get("AIVEN_AUTH_TOKEN")
+AIVEN_CA_CERT: Optional[str] = os.environ.get("AIVEN_CA_CERT")
+AIVEN_CLIENT_CONFIG: Optional[str] = os.environ.get(
+    "AIVEN_CLIENT_CONFIG", os.path.join(AIVEN_CONFIG_DIR, "aiven-client.json")  # type: ignore
```

Comment:
Got some errors here and could not figure it out:

```
aiven/client/envdefault.py:24: error: Argument 1 to "join" has incompatible type "Optional[str]"; expected "Union[str, PathLike[str]]"
```
 I have tried: using `Optional[str]`, `Union[str, PathLike[str]]` and `cast()` to `str`.

related to this maybe: https://stackoverflow.com/questions/60285688/python-typehint-for-os-getenv-causes-downstream-incompatible-type-errors

if someone has some ideas what to try here, I'd highly appreciate

Reply:
See the comment: https://github.com/aiven/aiven-client/pull/270/files#r800687065